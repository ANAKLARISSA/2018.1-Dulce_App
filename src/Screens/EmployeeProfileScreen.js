import React from 'react';
import {View , Text} from 'react-native';
import AGRButton from '../Components/AGRButton';

const styles = {

container: {
flex: 1,
flexDirection: 'column',
padding: 50,
backgroundColor:'#FFF'
},
text: {
fontSize: 36,
marginBottom: 15,
alignSelf: 'center'
},
informacoes: {
  alignSelf: 'center'
}

}

class EmployeeProfileScreen extends React.Component{

  constructor(props){
    super(props);
    this.state = {
      employee: [],
      loading: true

    }

  }

  componentDidMount(){

    const url = 'https://jsonplaceholder.typicode.com/users';
    this.setState({loading: true});
    fetch(url)
    .then((response) => {return response.json();})
    .then((json) => {this.setState({employee: json,loading: false});})
  }

  render(){
    return(
      <View style={styles.container}>
      {
        this.state.loading ? (<Text style={styles.text}>Carregando...</Text>) : (
          <View style={styles.informacoes}>
          <Text style={styles.text}>{this.state.employee[0].name}</Text>
          <Text style={styles.text}>CRM/Matricula:</Text>
          <Text style={styles.text}>{this.state.employee[0].address.zipcode}</Text>
          <Text style={styles.text}>setor: Pediatria</Text>
          <Text style={styles.text}>Cirurgi√£o</Text>
          <Text style={styles.text}>Hospital do Gama</Text>
          <View style={{height: 50}} />
          <AGRButton
          text='editar'
          onPress={() => this.props.navigation.navigate('edit')} />
          </View>


        )

      }
      </View>
    )

  }



}

export default EmployeeProfileScreen;
