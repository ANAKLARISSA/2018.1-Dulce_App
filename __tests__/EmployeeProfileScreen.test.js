import React from 'react';
import 'react-native';
import EmployeeProfileScreen from '../src/Screens/EmployeeProfileScreen';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
  .create(<EmployeeProfileScreen />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
