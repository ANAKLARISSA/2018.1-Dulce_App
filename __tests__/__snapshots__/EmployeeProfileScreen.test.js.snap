// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
      "flexDirection": "column",
      "padding": 50,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 30,
        "marginBottom": 15,
      }
    }
  >
    Carregando...
  </Text>
</View>
`;
